src/internal/converters/eml/eml.go

```diff
@@ -50,6 +51,92 @@ func formatAddress(entry models.EmailAddressable) string {
 	return fmt.Sprintf(addressFormat, name, email)
 }
 
+// getICalData converts the emails to an event so that ical generation
+// can generate from it.
+func getICalData(ctx context.Context, data models.Messageable) (string, error) {
+	msg := data.(*models.EventMessageRequest)
+	event := models.NewEvent()
+
+	// This method returns nil if data is not pulled using the necessary expand property
+	// .../messages/<message_id>/?expand=Microsoft.Graph.EventMessage/Event
+	// Also works for emails which are a result of someone accepting an invite
+	// If we add this field on a cancellation mail, the request fails.
+	// It look to be OK to run when listing emails but it gives empty({}) event value for cancellations.
+	// TODO(meain): cancelled event details are available when pulling .eml, figure out if we can get it too
+	mevent := msg.GetEvent()
+	if mevent != nil {
+		return ics.FromEventable(ctx, mevent)
+	}
+
+	// Exceptions(modifications) are covered under this, although graph just send the
+	// exception event and not the parent, which what eml also seems to contain
+	if ptr.Val(msg.GetMeetingMessageType()) != models.MEETINGREQUEST_MEETINGMESSAGETYPE {
+		// We don't have event data if it not "REQUEST" type.
+		// Both cancellation and acceptance does not return enough
+		// information to recreate an event.
+		return "", nil
+	}
+
+	// If data was not fetch through this, then we can approximate the
+	// details with the following
+	event.SetId(msg.GetId())
+	event.SetCreatedDateTime(msg.GetCreatedDateTime())
+	event.SetLastModifiedDateTime(msg.GetLastModifiedDateTime())
+	event.SetIsAllDay(msg.GetIsAllDay())
+	event.SetStart(msg.GetStartDateTime())
+	event.SetEnd(msg.GetEndDateTime())
+	event.SetRecurrence(msg.GetRecurrence())
+	// event.SetIsCancelled()
+	event.SetSubject(msg.GetSubject())
+	event.SetBodyPreview(msg.GetBodyPreview())
+	event.SetBody(msg.GetBody())
+
+	// https://learn.microsoft.com/en-us/graph/api/resources/eventmessage?view=graph-rest-1.0
+	// In addition, Outlook automatically creates an event instance in
+	// the invitee's calendar, with the showAs property as tentative.
+	event.SetShowAs(ptr.To(models.TENTATIVE_FREEBUSYSTATUS))
```

Comment:
not sure if this is worth pursuing, but given the comment above is this going to set the creator of the event to tentative as well? Is that what would normally happen for the person who created the event?

Reply:
Hmm, that is a tricky question. The only time it would cause an issue is if we export the "Sent Items" folder from the organizer -> convert to pst -> import them back. In this case, we would have incorrect status, but I'm not sure if it is worth worrying about. I don't know if exchange even parses out the emails "Sent Items" in imported pst files and create calendar events from them.


As far as the invitee, ie receiver is concerned, it is always "tentative" and the status of the owner is not affected by their personal cal.